sdk/ai/azure-ai-inference/azure/ai/inference/_patch.py

```diff
@@ -146,9 +147,159 @@ class ChatCompletionsClient(ChatCompletionsClientGenerated):
     """
 
     def __init__(self, endpoint: str, credential: Union[AzureKeyCredential, "TokenCredential"], **kwargs: Any) -> None:
-        self._model_info: Optional[_models.ModelInfo] = None
+        # First pop out any unlisted embedding settings, since we need to remove them before saving kwargs
+        model_info: Optional[_models.ModelInfo] = kwargs.pop("model_info", None)
```

Comment:
We should make these arguments keyword-only. We gain nothing by hiding them in `kwargs`. The main use case for `kwargs` in Python3 is when you want to pass through an unknown set of parameters (for example in a decorator for functions), or when you have an open set of parameter names (for example when creating a `dict`).

Reply:
@johanste I disagree. As I said before, exposing the same long list of input parameters on the constructor, on `.with_defaults()` and on `.complete()` is going to be very confusing and redundant. We need to pick one of two ways to support "sticky" settings: either expose them publicly on the constructor or expose them on a `.with_defaults()` method, if we want to keep the credential/transport settings separate from chat completions settings. But I don't think we should do BOTH. I would have not started this work on `.with_defaults()` if I knew it would lead to exposing chat completions settings on the constructor as well. If you want to abandon the `.with_default()` and set sticky chat completions settings on the constructor, we can discuss doing that work instead.

Reply:
We should do both. I still don't see what is confusing about this common idiom in python. I understand the argument that it may be redundant. But given that python doesn't have private/protected constructors, and the code you have written is functionally equivalent (although more error prone and obfuscates the signature enough to hide breaking changes from our tooling) to actually documenting the actual behavior. Please note that I would not consider it to be any different from a breaking change perspective if we were to stop supporting any of the parameters. They are still considered to be in the public API contract.

Reply:
@kristapratico resolving this since we decided not to ship a `with_defaults` method, but instead add chat completions/embedding settings in the constructor.